### PR TITLE
PR gateway: Added dependency between the groups to run in parallel/sequence

### PR DESCRIPTION
**Pull request by kparichay**
Solves issue #485, resolves problems reported in #492
Update over commit 9e583915855f579b0bc426d8d5e873ea379adf95
Solves circular name reference issue

1. Build test: [X]Passed [ ]Failed []Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

### CODE REVIEW

**Comment by leemgs** [comment]
I have tagged WIP and AGINGTEST to have a self assessment before a merging process. :)

**Comment by kparichay** [comment]
> I have tagged WIP and AGINGTEST to have a self assessment before a merging process. :)

Thanks.

File: ci/taos/checker-pr-gateway.sh [hunk 1]
```diff
@@ -135,34 +135,70 @@ $checkout_branch
...
+#  arg2: list of checker's name
+#  arg3: list of checker's log file
+#  arg@: all remaining arguments are to be passed to the checkers (common for all the checkers)
+function run_all_checkers(){
+  local -n cmd_list=$1
+  local -n name_list=$2
+  local -n logfile_list=$3
+  local checker_args=${@:4}
+
+  local num_checkers=${#cmd_list[@]}
```
Comment (leemgs) [inline ci/taos/checker-pr-gateway.sh]: > Solves circular name reference issue  #492.

Please, write a change by adding V4 in this commit body.
And, Could you keep a sequence (e.g., V4, V3, V2, V1) for a readability such as LKML mailling list ASAP. :)
For example,
V4 .....
V3 .....
V2 .....

Comment (kparichay) [reply ci/taos/checker-pr-gateway.sh]: updated.

File: ci/taos/checker-pr-gateway.sh [hunk 2]
```diff
@@ -135,34 +135,70 @@ $checkout_branch
...
+  for (( i=0; i<$num_checkers; i++ )); do
+    local logfile=${logfile_list[$i]}
+
+    cd $dir_ci
+    echo -e "[DEBUG] dir_commit is $dir_commit. This folder is created." | tee    $logfile
+    echo -e "[DEBUG] current path: $(pwd)."                              | tee -a $logfile
+    # Save a log file to debug the cheker
+    echo -e "[DEBUG] ${cmd_list[$i]} $checker_args "             | tee -a $logfile
```
Comment (leemgs) [inline ci/taos/checker-pr-gateway.sh]: Please keep the same location of "| tee -a ....." ASAP.

**Comment by leemgs** [comment]
I started an aging test to avoid unexpected situation such as CI failure after merging this PR.

File: ci/taos/checker-pr-gateway.sh (hunk 2 above)
Comment (kparichay) [reply ci/taos/checker-pr-gateway.sh]: updated

**Comment by leemgs** [comment]
The aging test is successfully done.

**Comment by leemgs** [comment]
LGTM

**Pull request merged**